### PR TITLE
feat: Feature/intmdb 1007 cluster outage

### PR DESCRIPTION
**Pull request by govardhanpagidi**
feat: intmdb-1007  cloud outage simulation cfn resource implementation

JIRA TIcket:  intmdb-1007 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [x] This change requires a documentation update

## Manual QA performed:

- [x] cfn invoke for each of CRUDL/cfn test
- [x] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [x] Published to AWS private registry
- [x] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [x] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code
- [ ] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

File: cfn-resources/cloud-outage-simulation/mongodb-atlas-cloudoutagesimulation.json
Comment (Adelmar92) [inline cfn-resources/cloud-outage-simulation/mongodb-atlas-cloudoutagesimulation.json:114]:
```suggestion
            "/properties/ClusterName",
            "/properties/Profile"
```

File: cfn-resources/cloud-outage-simulation/cmd/resource/model.go
Comment (Adelmar92) [inline cfn-resources/cloud-outage-simulation/cmd/resource/model.go:14]: ClusterType is not being used anywhere

File: cfn-resources/cloud-outage-simulation/cmd/resource/resource.go
Comment (Adelmar92) [inline cfn-resources/cloud-outage-simulation/cmd/resource/resource.go:263]:
```suggestion
		if resp.StatusCode ==  http.StatusNotFound  {
```

File: cfn-resources/cloud-outage-simulation/cmd/resource/resource.go
Comment (Adelmar92) [inline cfn-resources/cloud-outage-simulation/cmd/resource/resource.go:279]: we can get all of this status in a enum, lets not use the Utils enum, lets create e new one, inside this resource

File: cfn-resources/cloud-outage-simulation/cmd/resource/resource.go
Comment (andreaangiolillo) [inline cfn-resources/cloud-outage-simulation/cmd/resource/resource.go:101]: Where do you set the operation status to complete?

File: cfn-resources/cloud-outage-simulation/cmd/resource/resource.go
Comment (andreaangiolillo) [inline cfn-resources/cloud-outage-simulation/cmd/resource/resource.go:67]: Oh I see you set the success here.This is confusing since the name suggests that this should return only an error if the validation failed. Can we move the handler.ProgressEvent in this function?

File: cfn-resources/cloud-outage-simulation/mongodb-atlas-cloudoutagesimulation.json
Comment (rdharmedran) [reply cfn-resources/cloud-outage-simulation/mongodb-atlas-cloudoutagesimulation.json]: fixed

File: cfn-resources/cloud-outage-simulation/cmd/resource/resource.go
Comment (rdharmedran) [reply cfn-resources/cloud-outage-simulation/cmd/resource/resource.go]: fixed

File: cfn-resources/cloud-outage-simulation/cmd/resource/resource.go
Comment (rdharmedran) [reply cfn-resources/cloud-outage-simulation/cmd/resource/resource.go]: fixed

File: cfn-resources/cloud-outage-simulation/cmd/resource/model.go
Comment (rdharmedran) [reply cfn-resources/cloud-outage-simulation/cmd/resource/model.go]: fixed

File: cfn-resources/cloud-outage-simulation/cmd/resource/resource.go
Comment (rdharmedran) [reply cfn-resources/cloud-outage-simulation/cmd/resource/resource.go:67]: validateProgress method will validates status until it returns simulation success or failure and Respond with corresponding ProgressEvent.

**Review by Adelmar92 — APPROVED** [review]

**Comment by Adelmar92** [comment]
@rdharmedran LGTM, you can merge once you get Andrea's Approval

File: cfn-resources/cloud-outage-simulation/mongodb-atlas-cloudoutagesimulation.json
Comment (Adelmar92) [inline cfn-resources/cloud-outage-simulation/mongodb-atlas-cloudoutagesimulation.json:3]: the description still has the template description

**Comment by Adelmar92** [comment]
@rdharmedran the name of the resource is wrong, it should be ClusterOutrageSimulation, we need to change that

**Review by Adelmar92 — DISMISSED** [review]

**Review by andreaangiolillo — APPROVED** [review]

**Review by Adelmar92 — APPROVED** [review]

**Review by Adelmar92 — DISMISSED** [review]

**Review by Adelmar92 — APPROVED** [review]

**Pull request merged**
